### PR TITLE
Issue 2612 - hzn secretsmanager secret read command that allows authenticated users to see secret details

### PR DESCRIPTION
**Pull request by Isabel-Gan**
Signed-off-by: Isabel Gan <Isabel.Gan@ibm.com>

API changes: `LIST` method supported for listing multiple and individual secrets (for an individual secret, will return `{ exists: true/false }`. `GET` is supported only for individual secrets, and will return the secret details if the user is allowed to read the secret.

### CODE REVIEW

File: agreementbot/agreementworker.go [hunk 1]
```diff
@@ -852,7 +852,7 @@ func (b *BaseAgreementWorker) ValidateAndExtractSecrets(consumerPolicy *policy.P
 				}
 
 				// Call the secret manager plugin to get the secret details.
-				details, err := b.secretsMgr.GetSecretDetails(exchange.GetOrg(deviceId), secretUser, shortSecretName)
+				details, err := b.secretsMgr.GetSecretDetails(b.config.AgreementBot.ExchangeId, b.config.AgreementBot.ExchangeToken, exchange.GetOrg(deviceId), secretUser, shortSecretName)
```
Comment (dabooz) [inline agreementbot/agreementworker.go:855]: I think the more correct way is this:
GetSecretDetails(b.GetExchangeId(), b.GetExchangeToken(), ...)

File: agreementbot/governance.go [hunk 2]
```diff
@@ -251,7 +251,7 @@ func (w *AgreementBotWorker) GovernAgreements() int {
 											continue
 										}
 
-										details, err := w.secretProvider.GetSecretDetails(ag.Org, secretUser, secretName)
+										details, err := w.secretProvider.GetSecretDetails(w.EC.Id, w.EC.Token, ag.Org, secretUser, secretName)
```
Comment (dabooz) [inline agreementbot/governance.go:254]: I think the more correct way is this:
GetSecretDetails(w.GetExchangeId(), w.GetExchangeToken(), ...)

File: cli/cliutils/cliutils.go [hunk 3]
```diff
@@ -876,6 +876,89 @@ func AgbotGet(urlSuffix, credentials string, goodHttpCodes []int, structure inte
...
+	}
+
+	respBody := io.Reader(resp.Body)
+	if resp.Header.Get("Content-type") == "application/octet-stream" {
+		// Show progress of binary files downloading
+		msgPrinter.Print("Downloading object")
+		chunkNumber := 0
+		respBody = &progressReader{resp.Body, func(r int) {
+			chunkNumber++
+			if r > 0 {
```
Comment (dabooz) [inline cli/cliutils/cliutils.go:912]: Why do we need this for LIST? The response will never be very big. I expected this function to look a  lot more like the GET.

**Review by dabooz — CHANGES_REQUESTED** [review]

File: cli/cliutils/cliutils.go (hunk 3 above)
Comment (Isabel-Gan) [reply cli/cliutils/cliutils.go]: I just got rid of the progress-checking code; my code for AgbotGet uses ExchangeGet and there wasn't an existing ExchangeList so I took the code from ExchangeGet and switched out GET for LIST

**Review by dabooz — APPROVED** [review]

**Pull request merged**